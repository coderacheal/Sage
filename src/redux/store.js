import { configureStore } from '@reduxjs/toolkit';
import currencyReducer from './currencySlice';

const store = configureStore({
  reducer: {
    crypto: currencyReducer,
  },
});

export default store;
