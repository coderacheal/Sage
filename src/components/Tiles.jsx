import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { getCryptoData } from '../redux/currencySlice';

const Tiles = () => {
  const dispatch = useDispatch();
  const [search, setSearch] = useState('');
  const { crypto } = useSelector((store) => store.crypto);

  useEffect(() => {
    dispatch(getCryptoData());
  }, [dispatch]);

  const handleSearch = (e) => {
    setSearch(e.target.value);
  };

  const filtering = crypto.filter((coin) => (
    coin.name.toLowerCase().includes(search.toLowerCase())
  ));

  return (
    <div data-test-id="tiles">
      <h1>SMART CRYPTO</h1>
      <input type="text" className="search" value={search} placeholder="Search for a currency..." onChange={handleSearch} />
      <div className="allCryto">
        {filtering.map((eachCoin) => (
          <div key={eachCoin.id} className="individualCryptoTile">
            <img src={eachCoin.image.small} alt={eachCoin.name} />
            <p>{eachCoin.name}</p>
            <Link to={`/crypto/${eachCoin.id}`}>
              <i className="fa-solid fa-arrow-right" />
            </Link>
          </div>
        ))}
      </div>
    </div>
  );
};

export default Tiles;
