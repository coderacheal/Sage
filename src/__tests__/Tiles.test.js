import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import store from '../redux/store';
import Tiles from '../components/Tiles';

describe('CountryDetails', () => {
  it('should render the component', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Tiles />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
