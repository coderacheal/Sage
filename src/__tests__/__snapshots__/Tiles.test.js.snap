// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDetails should render the component 1`] = `
<div
  data-test-id="tiles"
>
  <h1>
    SMART CRYPTO
  </h1>
  <input
    className="search"
    onChange={[Function]}
    placeholder="Search for a currency..."
    type="text"
    value=""
  />
  <div
    className="allCryto"
  />
</div>
`;
